### ISSUE TITLE
Significant frame drop is occurred by HLS stream in multiple devices using playback speed at higher 1.0

### ISSUE THREAD

**Issue by dev-platong** [google/ExoPlayer#10328]
### ExoPlayer Version

2.17.1

### Devices that reproduce the issue

* fire tv stick 4K(AFTMM) running Fire OS 6.2.8.1
* SONY BRAVIA(KJ-43X8500H) running Android9

### Devices that do not reproduce the issue

* Pexel3 running Android 12

### Reproducible in the demo app?

Yes

### Reproduction steps

Reproduction at demo but Some changes is required so I create a branch at https://github.com/dev-platong/ExoPlayer/tree/lag-problem-research-apk

When you open demo app there are items labeled `Drop frame problem test streams` on the top of list.

You open it and choose any HLS streams. Then you can see drop frame. If you cannot recognize drop frame, please compare mp4 or dash version.

Result:

Dash 🆗 
https://boiling-anchorage-99045.herokuapp.com/reproduction_movies/not_reproduction.mp4

HLS 🆖 
https://boiling-anchorage-99045.herokuapp.com/reproduction_movies/reproduction.mp4

### Expected result

Do not drop frames.

### Actual result

Significant drop frames are occurred.

### Media

NOTE: reproduce synopsis using only hls streams.
I tested hls streams by shaka player in order to ensure not to drop frame originally.
Shaka player's result are fine (Able to play video without frame drops).
      
      {
        "name": "bit dash stream(hls original)",
        "uri": "https://bitdash-a.akamaihd.net/content/MI201109210084_1/m3u8s/f08e80da-bf1d-4e3d-8899-f0f6155f6efa.m3u8"
      },
      {
        "name": "bit dash stream(hls with ts by ffmpeg)",
        "uri": "https://boiling-anchorage-99045.herokuapp.com/playlist_ts.m3u8"
      },
      {
        "name": "bit dash stream(hls with fmp4 by ffmpeg)",
        "uri": "https://boiling-anchorage-99045.herokuapp.com/playlist_fmp4.m3u8"
      },
      {
        "name": "bit dash stream(mp4)",
        "uri": "https://boiling-anchorage-99045.herokuapp.com/bitdash.mp4"
      },
      {
        "name": "bit dash stream(DASH fmp4)",
        "uri": "https://boiling-anchorage-99045.herokuapp.com/manifest.mpd"
      },
      {
        "name": "bit dash stream(HLS with fmp4 without audio)",
        "uri": "https://boiling-anchorage-99045.herokuapp.com/no_audio/playlist_fmp4.m3u8"
      }

### Bug Report

- [X] You will email the zip file produced by `adb bugreport` to dev.exoplayer@gmail.com after filing this issue.

**Comment by dev-platong** [comment]
I sent bug report to dev.exoplayer@gmail.com.

**Comment by dev-platong** [comment]
Info:

If you play video by BRAVIA, frame drops are also reproduced by mp4 and dash.

If you play video by Fire TV stick 4K, frame drops is reproduced by hls only.

**Comment by ojw28** [comment]
> I tested hls streams by shaka player in order to ensure not to drop frame originally. Shaka player's result are fine (Able to play video without frame drops).

Were you playing at 2x speed with Shaka? If not, it's unclear whether this is relevant, unless you also see frame drops with ExoPlayer at 1x speed.

Over to @christosts who might have more input on the broader issue.

**Comment by dev-platong** [comment]
Sorry you are right! I've tested this problem at normal speed by shaka. Excuse me.

But I tested hls video by using [hls.js ](https://hls-js.netlify.app/demo/) at double speed, it works fine. So there have been still an issue.

**Comment by dev-platong** [comment]
More info:

I found [this post](https://www.reddit.com/r/bravia/comments/pjbypf/2x_speed_on_youtube/).

Maybe hardware restriction causes bravia's playback problem. Because mp4 and dash stream also have a drop frame symptoms.

But fire stick's problem maybe caused by HLS parser implementation. Because it works fine by mp4 and dash stream.

**Comment by tonihei** [comment]
This is a known issue that some TVs don't handle playback speeds != 1.0 very well. In particular, we know it's an issue for Sony Bravia and Amazon Fire TV sticks. (e.g. #7683, #8560 or #9773). Previous analyses showed that the problem seems to depend on the how the codec is configured and what media is being played. For example, it seems to be more prevalent for 50fps content. In any case, the problem seems to be on a very low level in the system where the TV is attempting to align frame release times to the screen, which only works for 1.0 speed playback.

We tried various configuration changes already to see if they help (see for example https://github.com/google/ExoPlayer/issues/9773#issuecomment-1018396726 for some suggestions). The only "fix" we found for the Amazon Fire TV stick specifically is force-setting an operating rate (see https://github.com/google/ExoPlayer/issues/9773#issuecomment-1004683089). This may also be the reason why you see a difference with HLS on this device, as other formats may set this flag already.

One promising idea is the suggestion in https://github.com/google/ExoPlayer/issues/7683#issuecomment-699036483, which we haven't tested yet. It looks like the devices are using the media timestamp set during decoding to decide when to show a frame and partly ignore the release time we specify (which includes the playback speed adjustment). The suggestion is to modify the media timestamps on their way through the decoder already so that the device can't use the original media timestamp anymore (or believes the speed adjusted timestamp is the original).

I'll mark as a duplicate of #7683 (as this seems to be closest equivalent).
